packages/energy-api-influxdb/src/reads/reads.service.ts

```diff
@@ -141,12 +142,23 @@ export class ReadsService implements OnModuleInit {
   private findByMeterQuery(meterId: string, filter: FilterDTO): string {
     return `
     from(bucket: "${this.bucket}")
-    |> range(start: ${filter.start}, stop: ${filter.end})
+    ${this.getRangeFilter(filter.start, filter.end)}
     |> limit(n: ${filter.limit}, offset: ${filter.offset})
     |> filter(fn: (r) => r.meter == "${meterId}" and r._field == "read")
     `;
   }
 
+  private getRangeFilter(start: string, end: string): string {
+    const endPlusOneMs = addMilliseconds(new Date(end), 1);
+
+    // Range doesn't include "end" into results for backward compatibility range is used
+    // and then dates are filtered by "filter" function to include "end" into results
+    return `
+      |> range(start: ${start}, stop: ${endPlusOneMs.toISOString()})
+      |> filter(fn: (r) => r._time >= ${start} and r._time <= ${end})
+    `;
+  }
+
```

Comment:
There is already e2e test for that. It's running fine on influxdb, but was failing on cloud influxdb. This function gets it fixed